jfr-microbench/src/main/java/io/netty/jfr/microbench/JfrHandlerBenchmark.java

```diff
@@ -0,0 +1,168 @@
+import org.openjdk.jmh.infra.Blackhole;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+
+@Warmup(iterations = 5)
+@Measurement(iterations = 5)
+@State(Scope.Benchmark)
+@Fork(1)
+public class JfrHandlerBenchmark extends AbstractMicrobenchmark {
+
+    private static final ChannelHandler CONSUMING_HANDLER = new ChannelInboundHandlerAdapter() {
+        @Override
+        public void channelRegistered(ChannelHandlerContext ctx) throws Exception {
+            // NOOP
+        }
+
+        @Override
+        public void channelUnregistered(ChannelHandlerContext ctx) throws Exception {
+            // NOOP
+        }
+
+        @Override
+        public void channelActive(ChannelHandlerContext ctx) throws Exception {
+            // NOOP
+        }
+
+        @Override
+        public void channelInactive(ChannelHandlerContext ctx) throws Exception {
+            // NOOP
+        }
+
+        @Override
+        public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception {
+            // NOOP
+        }
+
+        @Override
+        public void channelReadComplete(ChannelHandlerContext ctx) {
+            // NOOP
+        }
+
+        @Override
+        public void userEventTriggered(ChannelHandlerContext ctx, Object evt) throws Exception {
+            // NOOP
+        }
+
+        @Override
+        public void channelWritabilityChanged(ChannelHandlerContext ctx) throws Exception {
+            // NOOP
+        }
+
+        @Override
+        public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws Exception {
+            // NOOP
+        }
+
+        @Override
+        public boolean isSharable() {
+            return true;
+        }
+    };
+
+    private static final Exception exception = new RuntimeException("");
+
+    @Param({
+            "", "io.netty.channel.ChannelDuplexHandler", "io.netty.handler.logging.LoggingHandler",
+            "io.netty.jfr.JfrChannelHandler"
+    })
+    public String channelHandlerClassName;
+
+    private ChannelPipeline pipeline;
+
+    @Setup(Level.Iteration)
+    public void setup()
+            throws ClassNotFoundException, NoSuchMethodException, IllegalAccessException, InvocationTargetException,
+                   InstantiationException {
+        pipeline = new EmbeddedChannel().pipeline();
+        if (!channelHandlerClassName.isEmpty()) {
+            final Class<?> channelHandlerClass = Class.forName(channelHandlerClassName);
+            Object channelHandlerInstance;
+            final Constructor<?> constructor = channelHandlerClass.getConstructor();
+            channelHandlerInstance = constructor.newInstance();
+            pipeline.addLast((ChannelHandler) channelHandlerInstance);
+        }
+        pipeline.addLast(CONSUMING_HANDLER);
+    }
+
+    @TearDown
+    public void tearDown() {
+        pipeline.channel().close();
+    }
+
+    @Fork(jvmArgsPrepend = "-XX:-FlightRecorder")
```

Comment:
Not sure if it matters, but -XX:FlightRecorder option is going away. I think it is already deprecated.

Reply:
It's only for the benchmark, which I have been running with 11.